python/amici/sbml_import.py

```diff
@@ -203,55 +208,66 @@ def loadSBMLFile(self, SBMLFile):
 
         self.SBMLreader = sbml.SBMLReader()
         self.sbml_doc = self.SBMLreader.readSBML(SBMLFile)
-
-        if (self.sbml_doc.getNumErrors() > 0):
-            raise SBMLException('Provided SBML file does not exists or is invalid!')
+        self.checkLibSBMLErrors()
 
         # apply several model simplifications that make our life substantially easier
         if len(self.sbml_doc.getModel().getListOfFunctionDefinitions())>0:
             convertConfig = sbml.SBMLFunctionDefinitionConverter().getDefaultProperties()
-            status = self.sbml_doc.convert(convertConfig)
-            if status != sbml.LIBSBML_OPERATION_SUCCESS:
-                raise SBMLException('Could not flatten function definitions!')
+            self.sbml_doc.convert(convertConfig)
```

Comment:
I have not yet figured out what the return code actually is (it might be the total number of errors that occured (in that function call?)). If an error occurs it is guaranteed to end up in the sbml model errors and will remain accessible after other function calls/additional errors. Thus it should be sufficient to check errors only twice in `loadSBMLFile`. I will also document this in the code.

Reply:
Ah, so for every non-zero return value there is also a warning/error created? Then it's fine. 

I guess the actual error code is not too important, only `!= sbml.LIBSBML_OPERATION_SUCCESS`

Reply:
I think the non-zero return value only indicates is the index of the latest error. I am just not sure whether it is global or function specific.

Reply:
For example, in the steady state example the validateSBML call issues many warnings, but the return code will just be the index to one of them. So the return code is not actually an error code, but simply the index to a specific entry in the error log. So using sbml.LIBSBML_OPERATION_SUCCESS is a bit misleading as the only valid option is setting it to 0.